src/screens/Profile.jsx

```diff
@@ -1,41 +1,53 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
 
 import Card from "../components/Card";
-import styles from "./Profile.module.css";
-import { labs } from "../mocks";
+import styles from "./Profile.module.css"
 import Tags from "../components/Tags";
+import Loader from "../components/Loader"
+
+import { getDataById } from "../services/Services"
 
 const Profile = (props) => {
-  const lab = labs.find((lab) => lab.id === parseInt(props.match.params.id));
-
-  return (
-    <>
-      <div className={styles.profile}>
-        <div className={styles.hero}>
-          <Card imgUrl={lab.url}></Card>
-        </div>
-
-        <div className={styles.rightSide}>
-          <h3 className={styles.name}>{lab.name}</h3>
-
-          <section className={styles.skills}>
-            <p className={styles.tagsTitle}>Technologies</p>
-            {lab.technologies && <Tags tags={lab.technologies}></Tags>}
-          </section>
-
-          <section className={styles.objectives}>
-            <h4 className={styles.heading}>Objectives</h4>
-            <p>{lab.objectives}</p>
-          </section>
-
-          <section className={styles.description}>
-            <h4 className={styles.heading}>About</h4>
-            {lab.description}
-          </section>
-        </div>
-      </div>
-    </>
-  );
+  const [data, setData] = useState([])
+
+  useEffect(() => {
+    props.location.dataType === "labs" ?
+      getDataById("labs", props.match.params.id).then(res => {
+        setData(
+          <>
+            <div className={styles.profile}>
+              <div className={styles.hero}>
+                <Card imgUrl={res.company.profile_image}></Card>
+              </div>
+
+              <div className={styles.rightSide}>
+                <h3 className={styles.name}>{res.name}</h3>
+
+                <section className={styles.skills}>
+                  <p className={styles.tagsTitle}>Technologies</p>
+                  {res.technologies && <Tags tags={res.technologies}></Tags>}
+                </section>
+
+                <section className={styles.objectives}>
+                  <h4 className={styles.heading}>Objectives</h4>
+                  <p>{res.objectives}</p>
+                </section>
+
+                <section className={styles.description}>
+                  <h4 className={styles.heading}>About</h4>
+                  {res.company.description}
+                </section>
+              </div>
+            </div>
+          </>)
+      })
+      :
+      setData(<span>{"This is a candidate."}</span>)
```

Comment:
Same comment as above. Try to make the component "dumb". It should only have one template. No matter if it is a candidate or a lab, it should look the same- just different information.

That means the template doesn't change, just the data in it.

Try implementing the view for candidates as well!

Afterwards, try merging the Candidate profile with the Lab profile. Your `getDataById` method could make sure its `res` will always have the same fields for both types!